File: programming-skills-study-plan-ii/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        
        first, second = 1, 2
        for _ in range(2, n):
            tmp = second
            second = first + second
            first = tmp
        
        return second

File: cpp/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 2) return n;

        int i = 1;
        int j = 2;

        for (int k = 0; k < n - 2; k++) {
            int tmp = i + j;
            i = j;
            j = tmp;
        }

        return j;
    }
};